File: android/app/src/main/java/com/pan3d/display/particle/ball/ParticleBallGpuData.java
package com.pan3d.display.particle.ball;

import com.pan3d.display.particle.ParticleGpuData;
import com.pan3d.scene.Scene3D;

import java.nio.FloatBuffer;
import java.util.List;

public class ParticleBallGpuData extends ParticleGpuData {

    public Float[] basePos;
    public FloatBuffer basePosBuffer;
    public FloatBuffer speedBuffer;
    public List randomColor;
    public FloatBuffer randomColorBuffer;
    public FloatBuffer randomOffset;
    public List baseRotation;
    public FloatBuffer baseRotationBuffer;

    public ParticleBallGpuData(Scene3D val) {
        super(val);
    }

//    @property (nonatomic, assign)  GLfloat*    basePos;
//    @property (nonatomic, assign) GLuint   basePosBuffer;
//    @property (nonatomic, assign) GLuint   speedBuffer;
//    @property (nonatomic, strong)  NSMutableArray*   randomColor;
//    @property (nonatomic, assign) GLuint  randomColorBuffer;
//    @property (nonatomic, assign)  float   randomOffset;
//    @property (nonatomic, strong)  NSMutableArray*   baseRotation;
//    @property (nonatomic, assign) GLuint   baseRotationBuffer;
}

File: new/src/Pan3d/display/particle/ball/ParticleBallGpuData.ts
module Pan3d {
    export class ParticleBallGpuData extends ParticleGpuData {
        public basePos: Array<number>;
        public basePosBuffer: WebGLBuffer;

        public beMove: Array<number>;
        public beMoveBuffer: WebGLBuffer

        public randomColor: Array<number>;
        public randomColorBuffer: WebGLBuffer;
        public randomOffset: number;

        public baseRotation: Array<number>;
        public baseRotationBuffer: WebGLBuffer;

    }
}